src/main/scala/chiseltest/internal/ThreadedBackend.scala

```diff
@@ -877,4 +881,88 @@ class ThreadedBackend(annotations: AnnotationSeq) extends BackendInterface with
     }
     processThread(currentThread.get)
   }
+
+  def getClock(clk: Clock): Boolean =
+    simulatorInterface.peek(dataNames(clk)) match {
+      case Some(x) if x == BigInt(1) => true
+      case _                         => false
+    }
```

Comment:
As mentioned in a previous phase, these should fail noisily. Similarly below.

Reply:
I'll add an additional API to enable/disable this.
In the future, it's possible to peek a non-exist(constant-propagated) target inside Module.

Reply:
Can you give me a use case where you would want to peek a non-existent target and where some arbitrary result would be useful?

Like, I can see the utility of maybe `peekOption -> Option[BigInt]`, and punt to the upper layer to deal with what happens on a None / nonexistent wire, but I don't think (and I feel strongly about this, unless you have solid arguments the other way) anything should ever create results out of thin air, because this kind of non-intuitive data (even if expedient now) can lead down a fruitless debugging rabbit hole.

And even with peekOption, I'm not sure what would be guidelines for when to use it vs. when not to, and I'd prefer for APIs not to be introduced until we have a reasonable use case for them.

I also feel strongly against a stateful API to toggle this kind of behavior.

Reply:
For example: a API design here:
https://github.com/sequencer/diplomatictester/blob/4284f6919426186cc4d3e752bb0578d3104ce3cd/src/Driver.scala#L99
But `corrupt` is eliminated during constant propagation.
And if it directly fails:
https://github.com/sequencer/diplomatictester/blob/4284f6919426186cc4d3e752bb0578d3104ce3cd/tests/src/dutIOSpec.scala#L46
won't be able to run.

Reply:
The key issue of this is: If we are peeking/poking a constant-propagated target, what behavior should be?
Or we can `dontTouch` those signal explicitly, and throw it like you said.

Reply:
I would say the behavior should be to crash. Sure, a zero might work in the use case you have now, but might not work in another use case - so it's a bad idea to hardcode values created out of nothing. With a crash, at least it's noisy and the user gets a stack trace indicating what failed, and can add the needed dontTouch annotations. Compared to creating values - the user might just see a failed test, with an assertion that might trigger until much later (so nothing hinting at the true root cause), with RTL that is actually good, and might not even think about needing the debug the test infrastructure itself.